R/simulate_design.R

```diff
@@ -122,9 +122,9 @@ simulate_single_design <- function(design, sims) {
     results_list <- fan_out(design, sims)
     fan_id <- setNames(
       lapply(rev(sims$n), seq),
-      paste0("fan_", seq_len(nrow(sims)))
+      paste0("fan_", rev(seq_len(nrow(sims))))
     )
-    fan_id <- expand.grid(fan_id)
+    fan_id <- expand.grid(rev(fan_id))
```

Comment:
I see this:

```
> sims <- data.frame(end=1:3, n=c(1,5,2))
>     fan_id <- setNames(
+       lapply(rev(sims$n), seq),
+       paste0("fan_", rev(seq_len(nrow(sims))))
+     )
>     fan_id <- expand.grid(rev(fan_id))
> fan_id
   fan_1 fan_2 fan_3
1      1     1     1
2      1     2     1
3      1     3     1
4      1     4     1
5      1     5     1
6      1     1     2
7      1     2     2
8      1     3     2
9      1     4     2
10     1     5     2
> 
```


I think this is what actually matches fan_out().

```
> fan_id <- setNames(lapply(sims$n, seq), paste0("fan_", seq_len(nrow(sims))))
> rev(expand.grid(rev(fan_id)))
   fan_1 fan_2 fan_3
1      1     1     1
2      1     1     2
3      1     2     1
4      1     2     2
5      1     3     1
6      1     3     2
7      1     4     1
8      1     4     2
9      1     5     1
10     1     5     2
```

Should verify this, it needs to match fan_out() exactly.

Reply:
can you say more? I don't know what you mean.

Reply:
Shouldn't fan_3 be 1:10 (since each of them is a different execution)? Having it only 1,2 makes it look like it was only run twice.  See the example design I posted on the PR.

Reply:
Eg grouping by fan_3 would give you nonsense numbers I think.

Reply:
Sorry, I really don't understand.  if you were to group by step 2 in the example above, wouldn't it also give you nonsense?  i.e., the "1" group in step_2 refers to two independent draws, no?

Reply:
No, it's the same draw - estimand b in the design I posted was exactly the same.

Reply:
Here is a somewhat more explicit example where you can see how the random draws propogate:

```
> require(DeclareDesign)
> require(reshape2)
> 
> f1 <- local({i <- 0; function(){i<<- i+1; i} })
> f2 <- local({i <- 0; function(){i<<- i+1; i} })
> f3 <- local({i <- 0; function(){i<<- i+1; i} })
> 
> e1 <- declare_estimand(a=f1(), ar=rnorm(1), term=TRUE, label='e1')
> e2 <- declare_estimand(b=f2(), br=rnorm(1), term=TRUE, label='e2')
> e3 <- declare_estimand(c=f3(), cr=rnorm(1), term=TRUE, label='e3')
> out <- simulate_design(declare_population(sleep) + e1 + e2 + e3, sims=c(1,1,5,2))
Warning message:
We recommend you choose a higher number of simulations than 1 for the top level of simulation. 
> 
> dcast(out, design_label+fan_1+fan_2+fan_3~estimand_label+term, value.var="estimand")
   design_label fan_1 fan_2 fan_3 e1_a       e1_ar e2_b         e2_br e3_c         e3_cr
1      design_1     1     1     1    1 1.516604121    1 -1.3027235383    1  0.3052041011
2      design_1     1     2     1    1 1.516604121    2 -0.2363384340    3 -0.4535625357
3      design_1     1     3     1    1 1.516604121    3 -0.1916714742    5  1.5182842803
4      design_1     1     4     1    1 1.516604121    4 -0.9283954334    7 -0.9957436604
5      design_1     1     5     1    1 1.516604121    5  0.2547567283    9  0.1836362585
6      design_1     2     1     1    1 1.516604121    1 -1.3027235383    2  0.8118381599
7      design_1     2     2     1    1 1.516604121    2 -0.2363384340    4 -0.5319524033
8      design_1     2     3     1    1 1.516604121    3 -0.1916714742    6 -2.4692229482
9      design_1     2     4     1    1 1.516604121    4 -0.9283954334    8 -0.6679191953
10     design_1     2     5     1    1 1.516604121    5  0.2547567283   10 -1.0457122748
```